tests/unit/lib/Any_unittest.cc

```diff
@@ -691,3 +691,65 @@ TEST(Any, lazy_cloneable_visitable)
 	EXPECT_FALSE(any.visitable());
 	EXPECT_THROW(any.visit(nullptr), std::logic_error);
 }
+
+TEST(AnyParameterProperties, old_api_default)
+{
+	AnyParameterProperties params = AnyParameterProperties();
+
+	EXPECT_EQ(
+	    params.get_model_selection(),
+	    EModelSelectionAvailability::MS_NOT_AVAILABLE);
+	EXPECT_EQ(
+	    params.get_gradient(), EGradientAvailability::GRADIENT_NOT_AVAILABLE);
+	EXPECT_FALSE(params.get_model());
+}
+
+TEST(AnyParameterProperties, new_api_default)
+{
+	AnyParameterProperties params = AnyParameterProperties();
+
+	EXPECT_TRUE(params.equals(ParameterProperties()));
+	EXPECT_FALSE(params.has_property(ParameterProperties::MODEL));
+	EXPECT_FALSE(params.has_property(ParameterProperties::GRADIENT));
+	EXPECT_FALSE(params.has_property(ParameterProperties::HYPER));
+}
+
+TEST(AnyParameterProperties, old_custom_ctor)
+{
+	AnyParameterProperties params = AnyParameterProperties(
+	    "test", EModelSelectionAvailability::MS_NOT_AVAILABLE,
+	    EGradientAvailability::GRADIENT_AVAILABLE, true);
+
+	EXPECT_EQ(params.get_description(), "test");
+	EXPECT_EQ(
+	    params.get_model_selection(),
+	    EModelSelectionAvailability::MS_NOT_AVAILABLE);
+	EXPECT_EQ(params.get_gradient(), EGradientAvailability::GRADIENT_AVAILABLE);
+	EXPECT_TRUE(params.get_model());
+
+	EXPECT_TRUE(params.has_property(ParameterProperties::MODEL));
+	EXPECT_TRUE(params.has_property(ParameterProperties::GRADIENT));
+	EXPECT_TRUE(params.equals(
+	    ParameterProperties::MODEL | ParameterProperties::GRADIENT));
+    EXPECT_FALSE(params.equals(ParameterProperties::MODEL));
```

Comment:
everything else should be false. And that should be tested. Not just these two, same reasoning as above